include/libtorrent/aux_/rtc_signaling.hpp

```diff
@@ -0,0 +1,173 @@
+/*
+
+Copyright (c) 2019, Paul-Louis Ageneau
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in
+      the documentation and/or other materials provided with the distribution.
+    * Neither the name of the author nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef TORRENT_RTC_SIGNALING_HPP_INCLUDED
+#define TORRENT_RTC_SIGNALING_HPP_INCLUDED
+
+#include "libtorrent/config.hpp"
+#include "libtorrent/error_code.hpp"
+#include "libtorrent/io.hpp"
+#include "libtorrent/io_context.hpp"
+#include "libtorrent/peer_id.hpp"
+#include "libtorrent/span.hpp"
+#include "libtorrent/time.hpp"
+
+#include <boost/asio/basic_waitable_timer.hpp>
+#include <boost/functional/hash.hpp>
+
+#include <functional>
+#include <memory>
+#include <optional>
+#include <queue>
+#include <vector>
+#include <chrono>
+
+namespace rtc {
+	class PeerConnection;
+	class DataChannel;
+}
+
+namespace libtorrent {
+
+class alert_manager;
+class torrent;
+
+namespace aux {
+
+struct rtc_stream_init;
+
+const int RTC_OFFER_ID_LEN = 16;
```

Comment:
this could be `constexpr`. I see it's only used once though, so it might not matter. But would you mind adding a comment on what this is? Is it something tweakable or something defined by the protocol?

Reply:
I doesn't seem to be defined by the protocol (which is only losely-defined really), it only defines the length of locally-generated offer IDs.